client/server/middleware/unsupported-browser.js

```diff
@@ -2,6 +2,23 @@ import config from '@automattic/calypso-config';
 import { addQueryArgs } from 'calypso/lib/url';
 import analytics from 'calypso/server/lib/analytics';
 
+/**
+ * The locales that are supported in exempting browser support checks.
+ */
+const SUPPORTED_LOCALES = [ 'en', ...config( 'magnificent_non_en_locales' ) ];
+
+/**
+ * The paths that are allowed even if the browser is unsupported.
+ */
+const ALLOWED_PATHS = [ 'browsehappy', 'themes', 'theme', 'calypso' ];
+
+/**
+ * Regular expression matching paths that are allowed even if the browser is unsupported.
```

Comment:
Nit: some of the comment feels redundant - specifically the "how/what". The "why" makes sense to keep.

```suggestion
 * We allow some paths even if the browser is unsupported.
```